internal/config/config.go

```diff
@@ -56,15 +63,61 @@ func GetStudioSecret() string {
 	return vCfg.GetString("speakeasy_studio_secret")
 }
 
+func GetWorkspaceAPIKey(orgSlug, workspaceSlug string) string {
+	keys := vCfg.Sub(workspaceKeysKey)
+
+	if keys != nil {
+		return keys.GetString(getWorkspaceKey(orgSlug, workspaceSlug))
+	}
+
+	return ""
+}
+
+func SetWorkspaceAPIKey(orgSlug, workspaceSlug, key string) error {
+	keys := vCfg.GetStringMapString(workspaceKeysKey)
+	if keys == nil {
+		keys = make(map[string]string)
+	}
+
+	keys[getWorkspaceKey(orgSlug, workspaceSlug)] = key
+
+	vCfg.Set(workspaceKeysKey, keys)
+
+	return save()
+}
+
+func GetAuthenticatedWorkspaces() []string {
+	return maps.Keys(vCfg.GetStringMapString(workspaceKeysKey))
+}
+
+func getWorkspaceKey(orgSlug, workspaceSlug string) string {
+	return fmt.Sprintf("%s@%s", orgSlug, workspaceSlug)
+}
+
 func SetStudioSecret(secret string) error {
 	vCfg.Set("speakeasy_studio_secret", secret)
 	return save()
 }
 
-func SetSpeakeasyAuthInfo(info core.SpeakeasyAuthInfo) error {
-	vCfg.Set("speakeasy_api_key", info.APIKey)
-	vCfg.Set("speakeasy_workspace_id", info.WorkspaceID)
-	vCfg.Set("speakeasy_customer_id", info.CustomerID)
+func SetSpeakeasyAuthInfo(ctx context.Context, info core.SpeakeasyAuthInfo) error {
+	// Keep speakeasy-self as default workspace
+	if vCfg.GetString("speakeasy_workspace_id") != "self" {
```

Comment:
There can be many workspaces with id "self" do you anticipate that being an issue